src/main/java/world/bentobox/bentobox/BStats.java

```diff
@@ -2,163 +2,221 @@
-        }));
-    }
-
-    /**
-     * Sends the "flags display mode" of all the online players.
-     * @since 1.6.0
-     */
-    private void registerFlagsDisplayModeChart() {
-        metrics.addCustomChart(new Metrics.AdvancedPie("flagsDisplayMode", () -> {
-            Map<String, Integer> values = new HashMap<>();
-
-            Bukkit.getOnlinePlayers().forEach(player -> {
-                Flag.Mode mode = plugin.getPlayers().getFlagsDisplayMode(player.getUniqueId());
+      )
+    );
+  }
+
+  /**
+   * Increases the count of islands that got create since the last "data get" request from BStats.
+   * @since 1.1
+   */
+  public void increaseIslandsCreatedCount() {
+    islandsCreatedCount++;
+  }
+
+  /**
+   * Sends the enabled addons (except GameModeAddons) of this server.
+   * @since 1.1
+   */
+  private void registerAddonsChart() {
+    metrics.addCustomChart(
+      new Metrics.AdvancedPie(
+        "addons",
+        () -> {
+          Map<String, Integer> values = new HashMap<>();
+          plugin
+            .getAddonsManager()
+            .getEnabledAddons()
+            .stream()
+            .filter(
+              addon ->
+                !(addon instanceof GameModeAddon) && addon.getDescription().isMetrics()
+            )
+            .forEach(addon -> values.put(addon.getDescription().getName(), 1));
+          return values;
+        }
+      )
+    );
+  }
+
+  /**
+   * Sends the enabled GameModeAddons of this server.
+   * @since 1.4.0
+   */
+  private void registerGameModeAddonsChart() {
+    metrics.addCustomChart(
+      new Metrics.AdvancedPie(
+        "gameModeAddons",
+        () -> {
+          Map<String, Integer> values = new HashMap<>();
+          plugin
+            .getAddonsManager()
+            .getGameModeAddons()
+            .stream()
+            .filter(gameModeAddon -> gameModeAddon.getDescription().isMetrics())
+            .forEach(
+              gameModeAddon -> values.put(gameModeAddon.getDescription().getName(), 1)
+            );
+          return values;
+        }
+      )
+    );
+  }
+
+  /**
+   * Sends the enabled Hooks of this server.
+   * @since 1.6.0
+   */
+  private void registerHooksChart() {
+    metrics.addCustomChart(
+      new Metrics.AdvancedPie(
+        "hooks",
+        () -> {
+          Map<String, Integer> values = new HashMap<>();
+          plugin
+            .getHooks()
+            .getHooks()
+            .forEach(hook -> values.put(hook.getPluginName(), 1));
+          return values;
+        }
+      )
+    );
+  }
+
+  /**
+   * Sends the "category" this server is in depending on how many players it has.
+   * @since 1.6.0
+   */
+  private void registerPlayersPerServerChart() {
+    metrics.addCustomChart(
+      new Metrics.SimplePie(
+        "playersPerServer",
+        () -> {
+          int players = Bukkit.getOnlinePlayers().size();
+          if (players <= 0) return "0"; else if (players <= 10) return "1-10"; else if (
```

Comment:
This looks quite uglier than the previous version.

Reply:
agree, this is unreadable...
But it is classic JS style.